src/transformers/modeling_flash_attention_utils.py

```diff
@@ -20,9 +20,18 @@
 import torch
 import torch.nn.functional as F
 
+from .integrations.deepspeed import (  # DeepSpeed seq parallelism (aka Ulysses)
+    is_deepspeed_available,
+    is_deepspeed_sp_enabled,
+)
 from .utils import is_flash_attn_2_available, is_flash_attn_greater_or_equal
 
 
+if is_deepspeed_available():
+    from deepspeed.sequence.layer import _SeqAllToAll
+    from deepspeed.utils import groups as ds_comm_groups
```

Comment:
Can we only import these function in the block where we use them. We don't want the user to implicitly import them when they don't use it + It will make transformers import slower.

Reply:
Thanks for pointing this out @SunMarc, I was torn between placing import at the top or within the block where it is needed. I decided to go with the former for performance reason. Recall that _flash_attention_forward_ would be called multiple times, this could lead to repeated imports which could (possibly) degrade performance. Code readerability and mantainability are the other (minor) reasons for my choice.

Reply:
Oh indeed ! Thx for explaining. A potential solution could be to import under the condition if is_deepspeed_sp_enabled() instead since we are only using these imports with that condition. WDTY ? Could you add a quick comment on why we decided to put the import at the top ?